### ISSUE TITLE
False spacing-between-declarations-with-annotations

### ISSUE THREAD

**Issue by cy6erGn0m** [pinterest/ktlint#970]
## Expected Behavior
spacing-between-declarations-with-annotations shouldn't trigger if there is a line delimiter between declaration and annotation

## Observed Behavior
spacing-between-declarations-with-annotations triggers twice

## Steps to Reproduce
```kotlin
private class A
@Deprecated("")
constructor(a: Int)
```

## Your Environment
* Version of ktlint used: *0.39.0*
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): *Gradle, kotlinter-gradle 3.2.0*
* Version of Gradle used (if applicable): *6.6.1*
* Operating System and version: *Macos*
